### PR TITLE
#5466 - feat: fetching search result.

### PR DESCRIPTION
**Pull request by DipakShrestha18**

### CODE REVIEW

File: src/components/SearchQueryFetch/SolrConnectorDemo/index.js
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrConnectorDemo/index.js]: This should fix it:

```javascript
onSubmit={(e) => this.onSubmit(e)}
```

Think it's not working because the onSubmit function (defined on line 15) needs the `event` parameter!

**Review by chriddell — COMMENTED** [review]
See comment re: onSubmit, should get it working 👍

**Comment by chriddell** [comment]
Is this ready for review @DipakShrestha18, or still working on it?

**Comment by DipakShrestha18** [comment]
@chriddell Not yet. But you can leave a comment.

File: src/components/SearchForm/SearchForm.tsx
Comment (creido-welly) [inline src/components/SearchForm/SearchForm.tsx]: label and input will need to be linked for accessibility using htmlFor and id respectively, even tho the control is wrapped

File: src/components/SearchForm/SearchFormInput.tsx
Comment (creido-welly) [inline src/components/SearchForm/SearchFormInput.tsx:21]: This input needs to have id="search" so label's htmlFor attribute links correctly

Comment (chriddell) [reply src/components/SearchForm/SearchFormInput.tsx:21]: This is part of something I wrote, which @DipakShrestha18 has rebased into his branch - I'll pick this up and fix

Comment (chriddell) [reply src/components/SearchForm/SearchFormInput.tsx:21]: I'll create a ticket and address in a separate PR

File: src/components/SearchForm/_search-form.scss
Comment (creido-welly) [inline src/components/SearchForm/_search-form.scss:11]: Lots of rules containing `rem(<value>)`. These should be replaced with rounded multiples of --space-unit wherever possible. Happy for this to happen later but all tech debt needs ticketing

File: src/components/SearchForm/_search-form.scss
Comment (creido-welly) [inline src/components/SearchForm/_search-form.scss:30]: `%btn-base` should do the job here

File: src/components/SearchForm/_search-form.scss
Comment (chriddell) [reply src/components/SearchForm/_search-form.scss:11]: Imagine this will cause inconsistencies in the look of components as we port their styles from https://github.com/wellcometrust/corporate, though?

Comment (creido-welly) [reply src/components/SearchForm/_search-form.scss:11]: Minor changes at most

By rounding I mean to numbers which are relatively clean multiples of 8 to 2 decimal places :)

**Review by creido-welly — APPROVED** [review]

**Review by creido-welly — CHANGES_REQUESTED** [review]
The connector needs documenting. If this is to be addressed later then leave a TODO comment with a ticket number ideally.

If it's work in progress then either create the PR as draft or add _WIP_ in the main heading

**Comment by DipakShrestha18** [comment]
@chriddell @creido-welly I'm going to merge this to develop unless if you have any other comments on it. 

Just to let you know that when I run type check against it throws some error. Not sure what can I do about it.

**Comment by chriddell** [comment]
Cheers @DipakShrestha18 - I'll have a look through before merging, so hold off a bit as there may be some tidy ups needed

File: src/components/SearchQueryFetch/SolrQueryFetchApp.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchApp.tsx]: Can we rename this to `SearchQueryFetch` to prevent confusion? 

(App should only really be used for the top-level component - ie something that renders the _entire_ app)

File: src/components/SearchQueryFetch/SolrQueryFetchApp.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchApp.tsx]: Needs one more indent

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: Should wrap this return in brackets for clarity

```js
return (
  ...
);
```

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: Add blank line at end of file

File: src/components/SearchQueryFetch/index.ts
Comment (chriddell) [inline src/components/SearchQueryFetch/index.ts]: Add blank line at end of file

File: src/components/SearchQueryFetch/SolrQueryFetchApp.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchApp.tsx]: Add blank line at end of file

File: src/components/SearchQueryFetch/SearchQueryFetch.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SearchQueryFetch.tsx]: This file's unnecessary - replace it's contents with those of _SolrQueryFetchApp_ and remove _SolrQueryFetchApp_.

**Comment by creido-welly** [comment]
@DipakShrestha18 for any type checking which is unfathomable create a ticket, add a TODO comment with that ticket reference and disable typescript for the item in question for now

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (creido-welly) [inline src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: use full names instead of abbreviations to make each property more readable

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: Add blankline between these two lines (to indicate separation of concerns)

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: Add blankline after `constructor()` function close

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: Add blankline after `onSubmit` function closes

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: Rename this file (and component) to `SearchQueryFetchConnector.tsx`

**Review by chriddell — CHANGES_REQUESTED** [review]
Mostly just tidy-ups/filename changes needed

File: src/components/SearchQueryFetch/SolrQueryFetchApp.tsx
Comment (chriddell) [inline src/components/SearchQueryFetch/SolrQueryFetchApp.tsx]: This file should be the main _SearchQueryFetch_ component - see above comment.

File: src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx
Comment (DipakShrestha18) [reply src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: This reflects the properties coming from solr.

Comment (creido-welly) [reply src/components/SearchQueryFetch/SolrQueryFetchConnector.tsx]: ah ok

**Comment by DipakShrestha18** [comment]
@chriddell @creido-welly I have made all the changes on the comments made by you guys. Let me know if anything else needs changing.

File: src/components/SearchQueryFetchConnector/index.ts
Comment (chriddell) [inline src/components/SearchQueryFetchConnector/index.ts]: Add trailing blank line at end of file

File: src/components/SearchQueryFetchConnector/SearchQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetchConnector/SearchQueryFetchConnector.tsx]: We don't really need this extra component, it's only rendering the SolrQueryFetch component so let's just remove it.

File: src/components/SearchQueryFetchConnector/SearchQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetchConnector/SearchQueryFetchConnector.tsx]: This should be exported, e.g.

`export class SolrQueryFetch extends React.component {`

It means we can then either do a named or default import later:

named import:

`import { SolrQueryFetch } from 'x.js'`

default import:

`import SolrQueryFetch from 'x.js'`

(both do the same thing)

File: src/components/SearchQueryFetchConnector/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetchConnector/SolrQueryFetchConnector.tsx]: Name of class should match filename

File: src/components/SearchQueryFetchConnector/SolrQueryFetchConnector.tsx
Comment (chriddell) [inline src/components/SearchQueryFetchConnector/SolrQueryFetchConnector.tsx]: This should use an "absolute" import, e.g.

`import SearchForm from 'SearchForm';`

https://medium.com/beqode/absolute-vs-relative-import-paths-nodejs-1e4efa65a7bb

**Review by chriddell — CHANGES_REQUESTED** [review]
Still taking account of individual comments below (renaming components to match filenames, remove bits of code etc.) I think we should rename and restructure the component folder.

Can we have:

```
SearchQueryFetch/
  - index.ts
  - SearchQueryFetch.tsx (formerly SearchQueryFetchConnector.tsx)
  - SearchQueryFetchConnector.tsx (formerly SolrQueryFetchConnector.tsx)
```

This means that the export statement in _components/index.ts_ will also need updating

**Comment by DipakShrestha18** [comment]
@chriddell Can you look at the PR again?

**Review by creido-welly — APPROVED** [review]

**Review by chriddell — APPROVED** [review]

**Comment by chriddell** [comment]
All good

**Pull request merged**
